src/register_hall_of_fame.c

```diff
@@ -0,0 +1,3067 @@
+        sub_0200E024(data->monPics[REGHOF_PIC_WHOLE_MON + i], 1.0f, 1.0f);
+        sub_0200E0FC(data->monPics[REGHOF_PIC_WHOLE_MON + i], GX_OAM_MODE_NORMAL);
+        sub_0200E0FC(data->monPics[REGHOF_PIC_WHOLE_MON_SHADOW + i], GX_OAM_MODE_XLU);
+        UnkImageStruct_SetSpriteVisibleFlag(data->monPics[REGHOF_PIC_WHOLE_MON_SHADOW + i], FALSE);
+    }
+    data->monPics[REGHOF_PIC_WHOLE_PLAYER] = SpriteRenderer_LoadResourcesAndCreateSprite(data->spriteRenderer, data->spriteGfxHandler, &sPicTemplates[18]);
+    data->monPics[REGHOF_PIC_WHOLE_PLAYER_SHADOW] = SpriteRenderer_LoadResourcesAndCreateSprite(data->spriteRenderer, data->spriteGfxHandler, &sPicTemplates[19]);
+    sub_0200DF98(data->monPics[REGHOF_PIC_WHOLE_PLAYER], 1);
+    sub_0200DF98(data->monPics[REGHOF_PIC_WHOLE_PLAYER_SHADOW], 1);
+    UnkImageStruct_SetSpriteVisibleFlag(data->monPics[REGHOF_PIC_WHOLE_PLAYER], FALSE);
+    UnkImageStruct_SetSpriteVisibleFlag(data->monPics[REGHOF_PIC_WHOLE_PLAYER_SHADOW], FALSE);
+
+    if (PlayerProfile_GetTrainerGender(data->args->profile) == TRAINER_FEMALE) {
+        sub_02070D84(TRAINERCLASS_PKMN_TRAINER_LYRA, 2, &sp2C);
+    } else {
+        sub_02070D84(TRAINERCLASS_PKMN_TRAINER_ETHAN, 2, &sp2C);
+    }
+    r4 = AllocFromHeap(HEAP_ID_REGISTER_HALL_OF_FAME, 0x1900);
+    sub_020143E0(sp2C.narcId, sp2C.ncbr_id, HEAP_ID_REGISTER_HALL_OF_FAME, &sp1C, r4);
+    RegisterHallOfFame_ReplaceSpriteChar(r4, NNS_G2dGetImageLocation(Sprite_GetImageProxy(data->monPics[REGHOF_PIC_WHOLE_PLAYER]->sprite), NNS_G2D_VRAM_TYPE_2DMAIN), 3200);
+    FreeToHeap(r4);
+
+    GfGfxLoader_GXLoadPal(sp2C.narcId, sp2C.nclr_id, GF_PAL_LOCATION_MAIN_OBJ, (enum GFPalSlotOffset)NNS_G2dGetImagePaletteLocation(Sprite_GetPaletteProxy(data->monPics[REGHOF_PIC_WHOLE_PLAYER]->sprite), NNS_G2D_VRAM_TYPE_2DMAIN), 0x20, HEAP_ID_REGISTER_HALL_OF_FAME);
+
+    data->monPics[REGHOF_PIC_WHOLE_TWINKLE] = SpriteRenderer_LoadResourcesAndCreateSprite(data->spriteRenderer, data->spriteGfxHandler, &sPicTemplates[20]);
+    UnkImageStruct_SetSpriteVisibleFlag(data->monPics[REGHOF_PIC_WHOLE_TWINKLE], FALSE);
+    RegisterHallOfFame_SetSilhouettePalette(data, REGHOF_PIC_WHOLE_PLAYER_SHADOW, REGHOF_SILHOUETTE_PAL_BLACK);
+    GfGfx_EngineATogglePlanes(GX_PLANEMASK_OBJ, GF_PLANE_TOGGLE_ON);
+}
+
+static void RegisterHallOfFame_AfterWholePartyView_UnloadSpriteRes(RegisterHallOfFameData *data) {
+    u32 i;
+    int tag;
+
+    for (i = 0; i < 15; ++i) {
+        RegisterHallOfFame_DeletePicByIndex(data, i);
+    }
+    SpriteGfxHandler_UnloadPlttObjById(data->spriteGfxHandler, 55522);
+    SpriteGfxHandler_UnloadCharObjById(data->spriteGfxHandler, 55520);
+    SpriteGfxHandler_UnloadCellObjById(data->spriteGfxHandler, 55515);
+    SpriteGfxHandler_UnloadAnimObjById(data->spriteGfxHandler, 55515);
+    SpriteGfxHandler_UnloadPlttObjById(data->spriteGfxHandler, 55520);
+    SpriteGfxHandler_UnloadCharObjById(data->spriteGfxHandler, 55519);
+    for (i = 0; i < 6; ++i) {
+        tag = 55512 + i;
```

Comment:
55512 seems like a base location, there should probably be a constant for it

Reply:
tag defines can be resolved at a later date